pynecone/var.py

```diff
@@ -748,3 +748,54 @@ def type_(self):
         if "return" in self.fget.__annotations__:
             return self.fget.__annotations__["return"]
         return Any
+
+
+class PcList(list):
+    """A custom list that pynecone can detect its mutation."""
+
+    def __init__(
+        self,
+        original_list: list,
+        reassign_field: Callable = lambda: None,
+    ):
+        """Initialize PcList.
+
+        Args:
+            original_list (list): The original list
+            reassign_field (function | None):
+                The method in the parent state to reassign the field.
+            field_name (str): The field name of the list in the parent state.
+        """
+        self._reassign_field = reassign_field
+
+        super().__init__(original_list)
+
+    def append(self, *args, **kargs):
+        """Append.
+
+        Args:
+            args: The args passed.
+            kargs: The kwargs passed.
+        """
+        super().append(*args, **kargs)
+        self._reassign_field()
+
+    def __setitem__(self, *args, **kargs):
+        """Set item.
+
+        Args:
+            args: The args passed.
+            kargs: The kwargs passed.
+        """
+        super().__setitem__(*args, **kargs)
+        self._reassign_field()
+
+    def __delitem__(self, *args, **kargs):
+        """Delete item.
+
+        Args:
+            args: The args passed.
+            kargs: The kwargs passed.
+        """
+        super().__delitem__(*args, **kargs)
+        self._reassign_field()
```

Comment:
Did I miss any methods? We have `append`, `__setitem__`, and `__delitem__` now

Reply:
We could add these too: https://www.w3schools.com/python/python_ref_list.asp (can be in a follow up)

Reply:
More methods are added in e2dbc362265d0f434180bbb81fc2fd6439965e6e

Reply:
Nice thanks